File: Backend/Together/Together.DataAccess/Entities/UserEvent.cs
namespace Together.DataAccess.Entities;

public class UserEvent
{
    public int UserEventId { get; set; }
    public string UserId { get; set; }
    public int SportId { get; set; }
    public int EventStatusId { get; set; }
    public int SportExperienceId { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public DateTime EventDate { get; set; }
    public string EventHour { get; set; }
    public string City { get; set; }
    public string Country { get; set; }
    public string? EventImageUrl { get; set; }
    
    public virtual UserInfo UserInfo { get; set; }
    public virtual Sport Sport { get; set; }
    public virtual EventStatus EventStatus { get; set; }
    public virtual SportExperience SportExperience { get; set; }
    public virtual ICollection<UserFavoriteEvent> UserFavoriteEvents { get; set; }
    public virtual ICollection<UserEventRequest> UserEventRequest { get; set; }
    public virtual ICollection<UserEventLocation> UserEventLocations { get; set; }
    public virtual ICollection<ChatRoom> ChatRooms { get; set; }
    public virtual ICollection<Notification> Notifications { get; set; }
    
}

File: FrontendReact/together-app/src/api/models/UserEvent.ts
import SportType from "../enums/SportType";
import EventStatus from "../enums/EventStatus";
import SportExperience from "../enums/SportExperience";
import IUserInfo from "./UserInfo";

export interface UserEvent {
    userEventId: number;
    userId: string;
    sportId: number;
    eventStatusId: number;
    sportExperienceId: number;
    title: string;
    description: string;
    eventDate: Date;
    eventHour: string,
    city: string;
    country: string;
    eventImageUrl: string;
    userInfo: IUserInfo;
}

function convertUserEventToEnum(sportId: number, eventStatusId: number, sportExperienceId: number){
    return {
        sport: SportType[sportId],
        eventStatus: EventStatus[eventStatusId],
        sportExperience: SportExperience[sportExperienceId]
    }
}

function splitDateToMonthName(date: Date){
    const monthNames = [
        "January",
        "February",
        "March",
        "April",
        "May",
        "June",
        "Julu",
        "August",
        "September",
        "October",
        "November",
        "December",
      ];
    return {
        month: monthNames[date.getMonth()],
    }
}

export { convertUserEventToEnum, splitDateToMonthName};